Feature/pop up tips

Loann12 opened the pull request:
## What

- Add a care-tips feature to Plant Info:
  - a themed Tips button displayed to the right of the plant name,
  - an AI-backed generateCareTips method in the repo that builds a prompt and calls Gemini,
  - showCareTips / dismissCareTips logic in the PlantInfoViewModel,
  - AlertDialog to display tips with a dedicated close button.
- Add test tags and tests unit + UI to cover happy/error flows.

## Why

- Give users quick, actionable care advice tailored to the plant’s Latin name and current health status without leaving the app.
- Help users take care of their plants.

## How
**UI:** `PlantInfoScreen.kt`
- Adds a Tips button next to the plant name, uses theme colors, adaptive horizontal gap, and an AlertDialog that shows the tips.
- Adds test tags: `TIPS_BUTTON, TIPS_DIALOG, TIPS_TEXT, TIPS_CLOSE_BUTTON.`

**ViewModel:** `PlantInfoViewModel.kt`
- Adds `LOADING_TIPS_PLACEHOLDER`, `showCareTips(...)`, and `dismissCareTips()`; handles coroutine calls and errors; logs errors.

**Repository:** `PlantsRepositoryBase.kt`
- Implements `generateCareTips(latinName, healthStatus)` that constructs the prompt and calls `plantDescriptionCallGemini(prompt`).
- Adds KDoc and debug logs plantCareTipsLog before/after the AI call and an error log on exception.

## Testing

**ViewModel tests add:** 
- `showCareTips_setsDialogAndLoadsTips`
- `dismissCareTips_hidesDialogButPreservesText`
- `showCareTips_handlesRepositoryException_andShowsFallback`

**Repo tests add:**
- `generateCareTips_callsGeminiAndReturnsResponse`
- `generateCareTips_returnsFallbackOnException`

**UI tests add:**
- `tipsButton_isDisplayed`
- `clickingTipsButton_showsDialogWithTipsText`
- `closingTipsDialog_hidesDialog`

## Screenshots 
<img width="300" height="659" alt="Capture d’écran 2025-11-15 à 13 24 50" src="https://github.com/user-attachments/assets/18248b90-b75c-4642-828e-767278758c4a" />
<img width="304" height="646" alt="Capture d’écran 2025-11-15 à 13 24 25" src="https://github.com/user-attachments/assets/2e836657-c0be-4697-a5b6-6c40dbdeffab" />

## Anything Else
Adapt the figma

kulkix commented on app/src/main/java/com/android/mygarden/ui/plantinfos/PlantInfoScreen.kt:
```diff
@@ -177,21 +182,36 @@ fun PlantInfosScreen(
               }
 
           // --- Name and Latin Name Section ---
```
Also Tips section.

kulkix commented on app/src/main/java/com/android/mygarden/ui/plantinfos/PlantInfoScreen.kt:
Clean code !

kulkix commented on app/src/main/java/com/android/mygarden/model/plant/PlantsRepositoryBase.kt:
```diff
@@ -310,6 +310,39 @@ abstract class PlantsRepositoryBase : PlantsRepository {
+   *   plant's condition.
+   * @return A plain-text string containing care tips generated by the AI, or a short error message
+   *   if generation failed.
+   */
```
The documentation is not the same as in `PlantsRepository` but it is the same function.

kulkix commented on app/src/main/java/com/android/mygarden/model/plant/PlantsRepositoryBase.kt:
Good implementation !

kulkix commented on app/src/main/java/com/android/mygarden/model/plant/PlantsRepositoryBase.kt:
```diff
@@ -310,6 +310,39 @@ abstract class PlantsRepositoryBase : PlantsRepository {
+    } catch (e: Exception) {
+      Log.e("plantCareTipsLog", "Error while generating care tips for $latinName", e)
+      return "Unable to generate care tips at this time."
+    }
```
Good use of try catch !

kulkix commented on app/src/main/java/com/android/mygarden/ui/plantinfos/PlantInfoViewModel.kt:
LGTM.

kulkix reviewed (changes requested):
The code is clean and well documented. The tests are strong and the line coverage is at 95.0%. But it is not ready to be merged yet. There is a SonarQube issue that should be fixed. When a plant is not recognized by the API it still gives you a tip but it has no specific informations about the plant. You should either hide the button when the boolean isRecognized in the PlantInfoUIState is false or making it display that it has no informations about the plant.

kulkix commented:
Also you could make the text of the tips shorter. This is not requested, it is a suggestion.

Loann12 commented:
Thanks for your review @kulkix .
For the issue about not giving tips when a plant isn’t recognized, I added a condition in the ViewModel that checks if the plant name is empty or “Unknown.” If so, a “Plant unknown” message now appears.
Regarding the tip text length, I agree it was a bit too long, so I updated the prompt to request shorter, more concrete tips that suggest an immediate action.

kulkix commented on app/src/main/java/com/android/mygarden/model/plant/Plant.kt:
LGTM !

kulkix commented on app/src/main/java/com/android/mygarden/ui/plantinfos/PlantInfoViewModel.kt:
```diff
@@ -112,4 +115,52 @@ class PlantInfoViewModel(
+
+      // If the plant's latin name equals the unknown sentinel, don't attempt to
+      // generate tips — show a fallback message instead.
+      if (latinName.equals(Plant.UNKNOWN_NAME, true)) {
```
👍
